dataplicity/m2m/bencode.py

```diff
@@ -60,12 +61,33 @@ def encode(obj):
 
     def add_encode(obj):
         if isinstance(obj, bytes):
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                append(six.b("{}:".format(len(obj))))
+                append(obj)
         elif isinstance(obj, text_type):
             obj = obj.encode('utf-8')
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                # it is safe to recurse here, because under PY3, the `obj`
+                # instance will be bytes, rather then str (note the encode()
+                # call above)
+                add_encode(obj)
         elif isinstance(obj, number_types):
-            append(b"i{}e".format(obj))
+            # please note: because of unicode_literals being used, we can't use
+            # six.b, for the simple reason that the python2 implementation of
+            # b is as following:
+            # def b(v):
+            #     return v
+            # this, combined with the unicode literals means that a string
+            # (like "41" - after formatting a number) would actually turn into
+            # an unicode.. and thus, a number after encoding would turn into
+            # unicode object. This is not what we want. However, because these
+            # are only digits, they can be safely encoded with latin-1, which
+            # would produce bytes (in both python2 and 3)
+            append("i{}e".format(obj).encode("latin-1"))
```

Comment:
A string containing only digits (which this branch covers) would look exactly alike, regardless of the encoding used. I assumed that by using ANSI encoding, the process would be somewhat quicker (only 256 combinations)